controller/instance_manager_controller.go

```diff
@@ -1584,6 +1603,190 @@ func (imc *InstanceManagerController) createInstanceManagerPodSpec(im *longhorn.
+	imc.instanceManagerMonitorMutex.Lock()
+	defer imc.instanceManagerMonitorMutex.Unlock()
+
+	backingImageMonitorName := types.GetBackingImageMonitorName(imName)
+	stopCh, ok := imc.instanceManagerMonitorMap[backingImageMonitorName]
+	if !ok {
+		return
+	}
+
+	select {
+	case <-stopCh:
+		// stopCh channel is already closed
+	default:
+		close(stopCh)
+	}
+}
+
+func (m *InstanceManagerMonitor) BackingImageMonitorRun() {
+	m.logger.Infof("Start SPDK backing image monitoring %v", m.Name)
+
+	ctx, cancel := context.WithCancel(context.TODO())
+	notifier, err := m.proxyClient.SPDKBackingImageWatch(ctx)
+	if err != nil {
+		m.logger.WithError(err).Errorf("Failed to get the notifier for monitoring")
+		cancel()
+		close(m.monitorVoluntaryStopCh)
+		return
+	}
+
+	defer func() {
+		m.logger.Infof("Stop monitoring spdk backing image %v", m.Name)
+		cancel()
+		m.StopMonitorWithLock()
+		close(m.monitorVoluntaryStopCh)
+	}()
+
+	go func() {
+		continuousFailureCount := 0
+		for {
+			if continuousFailureCount >= engineapi.MaxMonitorRetryCount {
+				m.logger.Errorf("Instance manager SPDK backing image monitor streaming continuously errors receiving items for %v times, will stop the monitor itself", engineapi.MaxMonitorRetryCount)
+				m.StopMonitorWithLock()
+			}
+
+			if m.CheckMonitorStoppedWithLock() {
+				return
+			}
+
+			_, err = notifier.Recv()
+			if err != nil {
+				m.logger.WithError(err).Error("Failed to receive next item in spdk backing image watch")
+				continuousFailureCount++
+				time.Sleep(engineapi.MinPollCount * engineapi.PollInterval)
+			} else {
+				m.lock.Lock()
+				m.updateNotification = true
+				m.lock.Unlock()
+				continuousFailureCount = 0
+			}
+		}
+	}()
+
+	timer := 0
+	ticker := time.NewTicker(engineapi.MinPollCount * engineapi.PollInterval)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			if m.CheckMonitorStoppedWithLock() {
+				return
+			}
+
+			needUpdate := false
+
+			m.lock.Lock()
+			timer++
+			if timer >= engineapi.MaxPollCount || m.updateNotification {
+				needUpdate = true
+				m.updateNotification = false
+				timer = 0
+			}
+			m.lock.Unlock()
+
+			if !needUpdate {
+				continue
+			}
+			if needStop := m.pollAndUpdateV2BackingImageMap(); needStop {
+				return
+			}
+		case <-m.stopCh:
+			return
+		}
+	}
+}
+
+func (m *InstanceManagerMonitor) pollAndUpdateV2BackingImageMap() (needStop bool) {
+	im, err := m.ds.GetInstanceManager(m.imName)
+	if err != nil {
+		if datastore.ErrorIsNotFound(err) {
+			m.logger.Warn("stop monitoring because the instance manager no longer exists")
+			return true
+		}
+		utilruntime.HandleError(errors.Wrapf(err, "failed to get instance manager %v for monitoring", m.Name))
+		return false
+	}
+
+	if im.Status.OwnerID != m.controllerID {
+		m.logger.Warnf("stop monitoring the instance manager on this node (%v) because the instance manager has new ownerID %v", m.controllerID, im.Status.OwnerID)
```

Comment:
```suggestion
		m.logger.Warnf("Stop monitoring the instance manager on this node (%v) because the instance manager has new ownerID %v", m.controllerID, im.Status.OwnerID)
```

Reply:
fixed.